### PR TITLE
Hackweek: Customer with analytics

### PR DESCRIPTION
**Pull request by atorresveiga**
### Why
We want to match the iOS experience and include a customer section in the Hub Menu
pe5pgL-4L1-p2

### Description
This PR adds the necessary logic to fetch and save the Customer data from the analytics endpoint. Additionally, it also updates the fetch of customers from the analytics method to persist the data to the new DB table and includes the `analyticsCustomerId` to the `WCCustomer` model to have access to this property when the customer information comes from the analytics endpoint.

### Testing
It is best to test these changes from the Woo PR.

### CODE REVIEW

File: tests/api/src/test/java/APITesting_WCGateway.java [hunk 1]
```diff
@@ -11,7 +12,7 @@
 import static io.restassured.RestAssured.oauth2;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.hasItems;
-
```
Comment (atorresveiga) [inline tests/api/src/test/java/APITesting_WCGateway.java:15]: The test failed with a `400` error code. I am not sure if this is connected to the issues we are experiencing with the old testing sites. I proceeded to ignore the test until I had more context p1718898683026799-slack-CC7L49W13

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/store/WCCustomerStore.kt [hunk 2]
```diff
@@ -208,14 +213,90 @@ class WCCustomerStore @Inject constructor(
                     AppLog.e(AppLog.T.API, "Error fetching customers from analytics: ${response.error.message}")
                     WooResult(response.error)
                 }
+
                 response.result != null -> {
+                    // Save customer locally
```
Comment (atorresveiga) [inline plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/store/WCCustomerStore.kt:218]: Save results to local DB

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/persistence/entity/CustomerFromAnalyticsEntity.kt [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package org.wordpress.android.fluxc.persistence.entity
+
+import androidx.room.Entity
+import org.wordpress.android.fluxc.model.LocalOrRemoteId
+
+@Entity(tableName = "CustomerFromAnalytics", primaryKeys = ["localSiteId", "id", "userId", "username"])
+data class CustomerFromAnalyticsEntity (
```
Comment (atorresveiga) [inline plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/persistence/entity/CustomerFromAnalyticsEntity.kt:7]: New table to save customers from analytics info locally

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/network/rest/wpcom/wc/customer/CustomerRestClient.kt [hunk 4]
```diff
@@ -164,4 +164,44 @@ class CustomerRestClient @Inject constructor(private val wooNetwork: WooNetwork)
 
         return response.toWooPayload()
     }
+
+    suspend fun fetchSingleCustomerFromAnalyticsByUserId(
```
Comment (atorresveiga) [inline plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/network/rest/wpcom/wc/customer/CustomerRestClient.kt:168]: We don't have API to get a single customer, so I'm using the filters for that as `wp-admin` does

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/model/customer/WCCustomerMapper.kt [hunk 5]
```diff
@@ -113,6 +113,7 @@ class WCCustomerMapper @Inject constructor() {
             shippingLastName = dto.name?.lastNameFromName() ?: ""
             shippingPostcode = dto.postcode ?: ""
             shippingState = dto.state ?: ""
+            analyticsCustomerId = dto.id
```
Comment (atorresveiga) [inline plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/model/customer/WCCustomerMapper.kt:116]: The `customer analytics` model does have a link to the `customer` model (userId), but it is tricky to get the `customer analytics` model for a `customer` model. I included this optional variable to keep the `analyticsCustomerId` value for customer information we map from the analytics endpoints.

**Review by ThomazFB — APPROVED** [review]
Tested alongside https://github.com/woocommerce/woocommerce-android/pull/11826#pullrequestreview-2154781527

**Pull request merged**